### ISSUE TITLE
fetching qnamaker answer received from qnamaker dialog using memory object

### ISSUE THREAD

**Issue by BeheraMadan** [microsoft/BotFramework-Composer#4926]
Hi,

We have added qnamaker dialog action to query qnaknowledge base as shown in below screenshot.

![image](https://user-images.githubusercontent.com/70739031/99811333-0d537380-2b6b-11eb-8499-2dd9ec074ce9.png)

We are getting qna answer from qnamaker dialog action. But how can we capture the qnamaker answer received from qnamaker dialog using memory object. We are able to capture the question asked by user using "turn.activity.text" but
is there any way in composer to capture the qnamaker answer which bot is sending to the user. In qnamaker trace there is nothing available to capture the data we want. We want to capture qnamaker answer received from qnamaker dialog using memory object so that we can implement different flow if the answer received is "Sorry, I did not find an answer."

![image](https://user-images.githubusercontent.com/70739031/99811382-23613400-2b6b-11eb-8242-85b585ce7f0e.png)

**Comment by BeheraMadan** [comment]
Hi,
I think there is some misunderstanding. I will explain issue again. 
We have some triggers(Intent Recognized) and an 'unknown intent'(none) trigger. If no intent is matched it will go to unknown intent trigger and in that unknown intent trigger we are using Qnamaker dialog action to connect to Qna knowledge base as shown in below screen shot.

![image](https://user-images.githubusercontent.com/70739031/100121652-fed4c700-2e9e-11eb-9e42-84a2469b97b2.png)


![image](https://user-images.githubusercontent.com/70739031/100120885-7b1ada80-2e9e-11eb-820b-0bdf734e63d2.png)

Now the thing is that  we are getting qna answer from qnamaker dialog. But we are unable to trace the parameters such as qna confidence score, qna matched question etc. We want to store this parameter in the bot. Also  how can we trace the qnamaker answer received from qnamaker dialog using memory object( in Emulator). We are able to trace the question asked by user using "turn.activity.text" but is there any way in composer to trace & store the qnamaker answer which bot is sending to the user. In qnamaker trace there is nothing available to capture the data we want. We want to trace and store qnamaker answer received from qnamaker dialog using memory object so that we can implement different flow based on the different answers received from qnamaker dialog.

![image](https://user-images.githubusercontent.com/70739031/100122669-a81bbd00-2e9f-11eb-941c-ce9df2eb4175.png)

**Comment by stevkan** [comment]
@BeheraMadan, apologies for the delay and thank you for your patience. I should have an update for you shortly and will keep you updated.

**Comment by dmvtech** [comment]
@BeheraMadan I'm not sure either. The QnAMaker dialog sure doesn't have anything built in to do this, as you have found.  I played with a bunch of approaches, such as:

- `turn.lastresult.text`
- `turn.intents[0]`

No luck yet. I'll continue to investigate and see if there is a way or if a limitation.

Would you be able to accomplish your goal using [Active Learning](https://docs.microsoft.com/en-us/azure/cognitive-services/qnamaker/how-to/use-active-learning?tabs=v1) instead of this approach?

**Comment by dmvtech** [comment]
@BeheraMadan Taking a look at the QnA Sample template helps (or it did for me). 

Within a **QnA Intentet recognized** trigger, you can use something _like_ the following: 
`turn.recognized.answers[0].questions`

**Comment by dmvtech** [comment]
Hi @BeheraMadan I hope the above got your issue resolved. Please let us know if you are still having issues getting it to work.

**Comment by BeheraMadan** [comment]
Hi @dmvtech,


You had mentioned that: Within a QnA Intentet recognized trigger, we can use something like the following:
turn.recognized.answers[0].questions

'QnA Intentet recognized trigger'  can we use to create QnAKnowledge base within the composer. 
But we are not creating QnAKnowledge base within the composer. We have already created QnAKnowledge base manually on the QnAMaker.ai portal and we want to connect  to this knowledgebase in the composer.

As part of our project requirement, now  we must use 'QnAMaker dialog' in composer to configure QnAKnowledge base in the bot. 
So, by using 'QnAMaker dialog'  can we trace what answer bot is displaying to the user for QnA question. If answer is 'Sorry I did not find an answer', we want to implement some different flow based on requirement. So we want to trace whether the answer from QnA Knowledge base  is 'Sorry I did not find an answer' or something different if we are using 'QnAMaker dialog' in composer.


Thanks..
